### PR TITLE
Decipher adjacent nodes

### PR DESCRIPTION
**Pull request by vincentcerri**
Adjacent nodes can be calculated based on distance between nodes. If the nodes are within a certain distance from each other, they will be considered adjacent and therefore can be traveled to. 

**Acceptance Criteria:**
 A text file with a list of all classrooms and their latitude and longitude coordinates.
 A list of edges can be created from either method
 The list of edges can be inserted within a graph
 The list of edges within the map can be retrieved

This will be necessary for working on issue #29

### CODE REVIEW

File: app/src/main/java/com/example/scoutconcordia/DataStructures/Graph.java [hunk 1]
```diff
@@ -8,24 +8,29 @@
...
+
 // Assumption is no two points are equivalent
 public class Graph
 {
     private int numberOfNodes;
     private Node[] nodes;
     private N_aryTree breathFirstSearchResults;
 
-    private class Node
+    public class Node
```
Comment (jaychenblue) [inline app/src/main/java/com/example/scoutconcordia/DataStructures/Graph.java]: Why was the inner Node class changed from private to public?

Can you explain any differences and your reasoning for this decision?

File: app/src/main/java/com/example/scoutconcordia/DataStructures/Graph.java [hunk 2]
```diff
@@ -8,24 +8,29 @@
...
     {
         private int id;
         private LatLng element;
         private LinkedList<Node> adjacencyList;
         private boolean traversed;
+        private int type;  // 0 is a class node, 1 is a hall node.
```
Comment (jaychenblue) [inline app/src/main/java/com/example/scoutconcordia/DataStructures/Graph.java]: What is a hall node? Is it simply a node between corridors and hallways?

File: app/src/main/java/com/example/scoutconcordia/DataStructures/Graph.java [hunk 3]
```diff
@@ -343,10 +456,63 @@ public static Graph addNodesToGraph(String[] contents)
...
+                    } else {
+                        returnMe.insertVertex((LatLng)current.getEle(), 1); //insert a hall node
+                    }
                 current = current.getNext();
             }
         }
         return returnMe;
     }
+
+    public static LatLng readClassCoordinate(String currentLine)
```
Comment (jaychenblue) [inline app/src/main/java/com/example/scoutconcordia/DataStructures/Graph.java]: Where does String currentLine come from? When is this method called and by which class?

File: app/src/main/java/com/example/scoutconcordia/DataStructures/Graph.java [hunk 4]
```diff
@@ -343,10 +456,63 @@ public static Graph addNodesToGraph(String[] contents)
...
+                    insertEdge(currentNode.getElement(), nodes[j].getElement());
+            }
+        }
+    }
+
+    // calculates the distance between 2 nodes in meters
+    public float calculateNodeDistance(LatLng node1, LatLng node2)
+    {
+        float[] distance = new float[1];
+        distanceBetween(node1.latitude, node1.longitude, node2.latitude, node2.longitude, distance);
```
Comment (jaychenblue) [inline app/src/main/java/com/example/scoutconcordia/DataStructures/Graph.java]: I can't seem the find where this method is. Where is distanceBetween() defined?

File: app/src/main/java/com/example/scoutconcordia/DataStructures/Graph.java (hunk 2 above)
Comment (vincentcerri) [reply app/src/main/java/com/example/scoutconcordia/DataStructures/Graph.java]: Yes a class node is any node that is positioned on a class room.
A hall node is any node that is positioned in the hallways. 
In a search we look at the starting class node and ending class node, and get the path between the 2 using only hall nodes.

File: app/src/main/java/com/example/scoutconcordia/DataStructures/Graph.java (hunk 3 above)
Comment (vincentcerri) [reply app/src/main/java/com/example/scoutconcordia/DataStructures/Graph.java]: This is a helper method that is used inside the addNodesToGraph method above. 
currentLine is passed in when reading from a file of Nodes

File: app/src/main/java/com/example/scoutconcordia/DataStructures/Graph.java (hunk 4 above)
Comment (vincentcerri) [reply app/src/main/java/com/example/scoutconcordia/DataStructures/Graph.java]: distanceBetween() is an imported method from the android.location.Location class. This method calculates the difference between 2 LatLng points in meters

File: app/src/main/java/com/example/scoutconcordia/DataStructures/Graph.java (hunk 1 above)
Comment (vincentcerri) [reply app/src/main/java/com/example/scoutconcordia/DataStructures/Graph.java]: I needed to loop through Graph nodes on line 254 in MapsActivity. If Nodes was private I was not able to do this.

File: app/src/main/java/com/example/scoutconcordia/DataStructures/Graph.java (hunk 3 above)
Comment (jaychenblue) [reply app/src/main/java/com/example/scoutconcordia/DataStructures/Graph.java]: Shouldn't helper methods be private?

Comment (vincentcerri) [reply app/src/main/java/com/example/scoutconcordia/DataStructures/Graph.java]: yes i will change them to private!

File: app/src/main/java/com/example/scoutconcordia/DataStructures/Graph.java [hunk 5]
```diff
@@ -190,6 +197,97 @@ public int removeVertex(LatLng removeMe)
...
+     //       currentPointsToCycle = newCurrentPointsToCycle;
+     //       newCurrentPointsToCycle = new LinkedList<LatLng>(new LatLng(0, 0));
+     //       LinkedList.Node treeNodeChild = currentTreeNode.getChildren().getHead();
+     //       for (int i = 0; i < currentTreeNode.getChildren().size(); i++)
+     //       {
+     //           treeNodeChild = treeNodeChild.getNext();
+     //       }
+     //   }
+     //   return null;
+    //}
```
[the remark this answers is not in the document]
Comment (vincentcerri) [reply app/src/main/java/com/example/scoutconcordia/DataStructures/Graph.java]: yes just removed it

File: app/src/main/java/com/example/scoutconcordia/Activities/MapsActivity.java [hunk 6]
```diff
@@ -179,23 +199,137 @@ public boolean onNavigationItemSelected(@NonNull MenuItem menuItem) {
...
+                Graph hall_8_floor = createGraph("encryptedhall8nodes");
+                //System.out.println(hall_8_floor.vertices().length);
+
+                // This is temporary to help in placing the markers for each floor
+                //for (LatLng vertices : hall_8_floor.vertices())
+                //{
+//
+//                    Marker polyMarker = mMap.addMarker(new MarkerOptions().position(vertices));
+//                    hall8floorMarkers.add(polyMarker); // add the marker to the list of markers
+//                }
```
[the remark this answers is not in the document]
Comment (vincentcerri) [reply app/src/main/java/com/example/scoutconcordia/Activities/MapsActivity.java]: will be removed later as we are still in the process of adding nodes for the other floors

**Pull request merged**